google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/changestream/ReadChangeStreamResumptionStrategy.java

```diff
@@ -0,0 +1,96 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub.changestream;
+
+import com.google.api.core.InternalApi;
+import com.google.api.gax.retrying.StreamResumptionStrategy;
+import com.google.bigtable.v2.ReadChangeStreamRequest;
+import com.google.bigtable.v2.ReadChangeStreamRequest.Builder;
+import com.google.bigtable.v2.StreamContinuationToken;
+import com.google.bigtable.v2.StreamContinuationTokens;
+import com.google.cloud.bigtable.data.v2.models.ChangeStreamRecordAdapter;
+
+/**
+ * An implementation of a {@link StreamResumptionStrategy} for change stream records. This class
+ * tracks the continuation token and upon retry can build a request to resume the stream from where
+ * it left off.
+ *
+ * <p>This class is considered an internal implementation detail and not meant to be used by
+ * applications.
+ */
+@InternalApi
+public class ReadChangeStreamResumptionStrategy<ChangeStreamRecordT>
+    implements StreamResumptionStrategy<ReadChangeStreamRequest, ChangeStreamRecordT> {
+  private final ChangeStreamRecordAdapter<ChangeStreamRecordT> changeStreamRecordAdapter;
+  private String token = null;
+
+  public ReadChangeStreamResumptionStrategy(
+      ChangeStreamRecordAdapter<ChangeStreamRecordT> changeStreamRecordAdapter) {
+    this.changeStreamRecordAdapter = changeStreamRecordAdapter;
+  }
+
+  @Override
+  public boolean canResume() {
+    return true;
+  }
+
+  @Override
+  public StreamResumptionStrategy<ReadChangeStreamRequest, ChangeStreamRecordT> createNew() {
+    return new ReadChangeStreamResumptionStrategy<>(changeStreamRecordAdapter);
+  }
+
+  @Override
+  public ChangeStreamRecordT processResponse(ChangeStreamRecordT response) {
+    // Update the token from a Heartbeat or a ChangeStreamMutation.
+    if (changeStreamRecordAdapter.isHeartbeat(response)) {
+      this.token = changeStreamRecordAdapter.getTokenFromHeartbeat(response);
+    }
+    if (changeStreamRecordAdapter.isChangeStreamMutation(response)) {
+      this.token = changeStreamRecordAdapter.getTokenFromChangeStreamMutation(response);
+    }
+    return response;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Given a request, this implementation will narrow that request to include data changes that
+   * come after {@link #token}.
+   */
+  @Override
+  public ReadChangeStreamRequest getResumeRequest(ReadChangeStreamRequest originalRequest) {
+    // A null token means that we have not successfully read a Heartbeat nor a ChangeStreamMutation,
+    // so start from the beginning.
+    if (this.token == null) {
+      return originalRequest;
+    }
+
+    Builder builder = originalRequest.toBuilder();
+    // We need to clear either start_time or continuation_tokens.
```

Comment:
Should it be
```suggestion
    // We need to clear start_time and continuation_tokens.
```
? Looks like both start from and continuation tokens are updated